packaging/release.py

```diff
@@ -1345,8 +1345,11 @@ def test_sdist() -> None:
                 sdist = stack.enter_context(tarfile.open(sdist_file))
             except FileNotFoundError:
                 raise ApplicationError(f"Missing sdist: {sdist_file.relative_to(CHECKOUT_DIR)}") from None
-
-            sdist.extractall(temp_dir)
+            if hasattr(tarfile, 'data_filter'):
+                # Remove this check when Python 3.11 is EOL
+                sdist.extractall(temp_dir, filter='data')
```

Comment:
Suppress the `mypy` type error since it doesn't understand the `hasattr` guard we're using:

```suggestion
                sdist.extractall(temp_dir, filter='data')  # type: ignore[call-arg]
```